iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -580,12 +588,14 @@ class string
     /// @note out of bounds access leads to program termination
     constexpr const char& operator[](const uint64_t pos) const noexcept;
 
+    // AXIVION Next Construct AutosarC++19_03-A11.3.1: It's Ok
     template <uint64_t N>
     friend class string;
 
+    // AXIVION Next Construct AutosarC++19_03-A11.3.1: It's Ok
```

Comment:
Here you could state that this is a factory method for the `cxx::string` and this doesn't break the encapsulation.

Reply:
@FerdinandSpitzschnueffler Sure.